riscv-runtime/src/hash.rs

```diff
@@ -14,7 +14,7 @@ const GOLDILOCKS: u64 = 0xffffffff00000001;
 /// Goldilocks field.
 pub fn poseidon_gl_unsafe(data: &mut [u64; 12]) -> &[u64; 4] {
     unsafe {
-        asm!("ecall", in("a0") data as *mut [u64; 12], in("t0") u32::from(Syscall::PoseidonGL));
+        asm!("ecall", in("a0") data as *mut [u64; 12], in("t0") u32::from(Syscall::Poseidon));
```

Comment:
Please leave it `PoseidonGL`. When we introduce an interface for `PoseidonBB`, we must not give the impression that it is the same function with the same results.

Reply:
This ends up in the RISCV BB machine too though, so what do we do there?

Reply:
Or well, the new `finalize` function calls this

Reply:
Panic as unimplemented.

If the user doesn't care about what specific hash function is being used, maybe there should be a "fn hash_native()" that redirects to whatever specialized hash implementation for that field.

Reply:
We have tons of tests for the RISCV BB machine that would panic that way, I don't think it's reasonable to just remove them.

Reply:
How do they work if we don't have poseidon on BB yet? It should only panic if called.

Reply:
In the current version of this PR it translates to the `poseidon` instruction which is a no-op internally.